docs/farming-&-staking/staking/operators.mdx

```diff
@@ -614,9 +666,93 @@ To run both operator and validator at the same time, provide requrired flags for
 --bootnodes /ip4/140.82.45.36/tcp/40333/p2p/12D3KooWJxB2PVzyNqZywoCWAMLGhKr2LCbFMXbAPzUPQZon7yt1 \\
 --bootnodes /ip4/167.179.116.135/tcp/40333/p2p/12D3KooWHsRae3ip2uQAUPtpMcStHGGw777SJT3nu9maFmtsGry7 `}
 </CodeBlock>
-
 </TabItem>
 
+<TabItem value="docker" label="🐋 Docker">
+
+<CodeBlock>
+{`version: "3.7"
+services:
+  node:
+    # Replace snapshot-DATE with the latest release (like snapshot-2023-dec-11)
+    # For running on Aarch64 add -aarch64 after DATE
+    image: ghcr.io/subspace/node:snapshot-DATE
+    volumes:
+# Instead of specifying volume (which will store data in /var/lib/docker), you can
+# alternatively specify path to the directory where files will be stored, just make
+# sure everyone is allowed to write there
+      - node-data:/var/subspace:rw
+#      - /path/to/subspace-node:/var/subspace:rw
+    ports:
+# If port 30333 or 30433 is already occupied by another Substrate-based node, replace all
+# occurrences of 30333 or 30433 in this file with another value
+      - "0.0.0.0:30333:30333/udp"
+      - "0.0.0.0:30333:30333/tcp"
+      - "0.0.0.0:30433:30433/udp"
+      - "0.0.0.0:30433:30433/tcp"
+    restart: unless-stopped
+    command: [
+# Make sure to replace your_domain_id and your_operator_id with your domain ID and operator ID
+# Adjust the keystore path to the location where you generated your operator keypair.
+      "--chain", "gemini-3g",
+      "--blocks-pruning", "256",
+      "--state-pruning", "archive",
+      "--no-private-ipv4",
+      "--validator",
+      "-- ",
+      "--domain-id", "your_domain_id"
+      "--operator-id", "your_operator_id",
+      "--keystore-path", "/keystore",
+      "--bootnodes", "/ip4/3.87.28.170/tcp/40333/p2p/12D3KooWGHtULvhdKMZtzigSK1438uWXPj9rBQHVzTaKMWv1WRXp",
+      "--bootnodes", "/ip4/140.82.45.36/tcp/40333/p2p/12D3KooWJxB2PVzyNqZywoCWAMLGhKr2LCbFMXbAPzUPQZon7yt1",
+      "--bootnodes", "/ip4/167.179.116.135/tcp/40333/p2p/12D3KooWHsRae3ip2uQAUPtpMcStHGGw777SJT3nu9maFmtsGry7",
+      "--listen-addr", "/ip4/0.0.0.0/tcp/40333",
+      "--rpc-cors", "all",
```

Comment:
I'm not 100% sure without testing but I think we'll need the `--rpc` flags on the validator side too or the farmer will not be able to connect. I also suspect we do not need them on the operator side unless the user wants to use the RPC to interact with the domain.

@nazar-pc do you know off the top of your head?

Reply:
Some of them would be necessary for healthchecks, but I don't think there are any and unsafe RPC methods are certainly not necessary.

Well, actually this whole file seems to be broken. `farmer:` is not formatted properly and these RPC arguments should actually be applied to consensus node, while they are moved to the bottom whewe they now apply to domain node instead. Just like `--name` is not applied to domain node instead of consensus node, so users will not find themselves in telemetry anymore.

Also why do we have IP addresses for bootstrap nodes instead of domain names? IP addresses can change any time.

Also `--listen-addr` should be specified for consensus node, but it wasn't.

Also `--keystore` should not be necessary here unless there is a very good reason for it, it will probably likely not have correct permissions to write to.

The whole thing is a bit of a mess right now. The arguments before `--` should be exactly the same as they are without operator. Operator arguments should be added after `--`. Did someone try to copy this snippet into `docker-compose.yml` and run it?